### PR TITLE
Add publish info to README

### PR DESCRIPTION
**Pull request by galactusss**
Copied from @0x6431346e PR #32 

An attempt to add some information about publishing (some one of which should be moved here: https://hack.aragon.org/docs/tutorial.html#publishing see https://github.com/aragon/hack/issues/33, ideally this should only be a summary).

Blockers:
aragon/aragon-cli#309
aragon/aragon-cli#310

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,11 +41,134 @@ Running your app using IPFS will mimic the production environment that will be u
 - **build:app**: Builds the front-end
 - **build:script**: Builds the background script
 - **test**: Runs tests for the contracts
-- **publish:minor**: Release a minor version to aragonPM
+- **publish:patch**: Release a patch version of the frontend to aragonPM
+- **publish:minor**: Release a minor version of the frontend to aragonPM
```
Comment (sohkai) [inline README.md]: For the above, I'd reword as "Release a patch/minor version to aragonPM (only frontend content changes allowed)"

File: README.md [hunk 2]
```diff
@@ -41,11 +41,134 @@ Running your app using IPFS will mimic the production environment that will be u
...
+| Environment   | Network   |
+|---            |---        |
+| default       | localhost |
+| staging       | rinkeby   |
+| production    | mainnet   |
+
+Prerequisites:
+- ENS Registry address
+
+Note: the `default` environment which points to `localhost` does not have an ENS Registry address specified, that's because the `@aragon/cli` will use the default value of `0xB9462EF3441346dBc6E49236Edbb0dF207db09B7`.
```
Comment (sohkai) [inline README.md]: Would reword as:

```
... because the `@aragon/cli` will default the value to `0xB9462EF3441346dBc6E49236Edbb0dF207db09B7` (the ENS Registry pre-deployed on the local development chain).
```

File: README.md [hunk 3]
```diff
@@ -41,11 +41,134 @@ Running your app using IPFS will mimic the production environment that will be u
...
 - **publish:major**: Release a major version to aragonPM with a potentially new contract address for on-chain upgrades
+- **versions**: Check the currently installed versions of the app
 
 ### Libraries
 
 - [**@aragon/os**](https://github.com/aragon/aragonos): Aragon interfaces
 - [**@aragon/client**](https://github.com/aragon/aragon.js/tree/master/packages/aragon-client): Wrapper for Aragon application RPC
 - [**@aragon/ui**](https://github.com/aragon/aragon-ui): Aragon UI components (in React)
+
+## Publish
```
Comment (sohkai) [inline README.md]: Small formatting tweak: we generally put a newline after the headers

File: README.md [hunk 4]
```diff
@@ -41,11 +41,134 @@ Running your app using IPFS will mimic the production environment that will be u
...
+
+Note: the `default` environment which points to `localhost` does not have an ENS Registry address specified, that's because the `@aragon/cli` will use the default value of `0xB9462EF3441346dBc6E49236Edbb0dF207db09B7`.
+
+### Introduction to environments
+Environments are defined in `arapp.json`, for example `staging` points to:
+- an ENS registry (`0x314159265dd8dbb310642f98f50c066173c1259b`)
+- an APM registry (`open.aragonpm.eth`)
+- an APM repository (`app`)
+- an Ethereum network (`rinkeby`)
+- an Ethereum websockets provider (for **reading** from the blokchain)
```
Comment (sohkai) [inline README.md]: ~~blokchain~~ blockchain

Also, I'm not 100% sure how the websockets is provided, but I believe this [`wsRPC`](https://github.com/aragon/aragon-cli/pull/270) instead?

File: README.md [hunk 5]
```diff
@@ -41,11 +41,134 @@ Running your app using IPFS will mimic the production environment that will be u
...
+
+the `rinkeby` network is further defined in `truffle.js`, and has:
```
Comment (sohkai) [inline README.md]: ~~the~~ The `rinkeby`

File: README.md (hunk 1 above)
Comment (0x6431346e) [reply README.md]: Please see https://github.com/aragon/aragon-cli/issues/310 I'm trying to figure out how major/minor/patch work, based on the aragonPM restrictions. Shouldn't it be possible to release a bugfix to the contract for example?

File: README.md (hunk 4 above)
Comment (0x6431346e) [reply README.md]: Yes, it's `wsRPC`, I made it a bit more explicit

**Comment by 0x6431346e** [comment]
> Looking good! Let's merge it once the blockers get finished :).

I will merge this now, since the blockers are "cleared" more or less.
I will continue with "how to install the app in a dao" in a separate PR.
